types/i18next-xhr-backend/index.d.ts

```diff
@@ -1,36 +1,74 @@
-// Type definitions for i18next-xhr-backend 1.2.0
+// Type definitions for i18next-xhr-backend 1.4
 // Project: https://github.com/i18next/i18next-xhr-backend
-// Definitions by: Jan Mühlemann <https://github.com/jamuhl>
+// Definitions by: Jan Mühlemann <https://github.com/jamuhl>, Giedrius Grabauskas <https://github.com/GiedriusGrabauskas>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare module 'i18next-xhr-backend' {
-
-    interface Interpolator {
-        interpolate: () => string
-    }
-    interface Services {
-        interpolator: Interpolator
-    }
+declare namespace I18NextXhrBackend {
+    type LoadPathOption = string | ((lngs: string[], namespaces: string[]) => string);
 
     interface BackendOptions {
-        loadPath?: string | Function,
-        addPath?:  string,
-        allowMultiLoading?: boolean,
-        parse?: Function,
-        crossDomain?: boolean,
-        withCredentials?: boolean,
-        ajax?: Function
-    }
+        /**
+         * path where resources get loaded from, or a function
+         * returning a path:
+         * function(lngs, namespaces) { return customPath; }
+         * the returned path will interpolate lng, ns if provided like giving a static path
+         */
+        loadPath?: LoadPathOption;
+        /**
+         * path to post missing resources
+         */
+        addPath?: string;
+        /**
+         * your backend server supports multiLoading
+         * locales/resources.json?lng=de+en&ns=ns1+ns2
+         * set loadPath: '/locales/resources.json?lng={{lng}}&ns={{ns}}' to adapt to multiLoading
+         */
+        allowMultiLoading?: boolean;
+        /**
+         * parse data after it has been fetched
+         * in example use https://www.npmjs.com/package/json5
+         * here it removes the letter a from the json (bad idea)
+         */
+        parse?(data: string): string;
+        /**
+         * allow cross domain requests
+         */
+        crossDomain?: boolean;
+        /**
+         * allow credentials on cross domain requests
+         */
+        withCredentials?: boolean;
+        /**
+         * define a custom xhr function
+         * can be used to support XDomainRequest in IE 8 and 9
+         */
+        ajax?(url: string, options: BackendOptions, callback: AjaxRequestCallback, data: {} | string, cache: boolean): void;
```

Comment:
`data` should probably be `object | string`, but that would require you to add a `// TypeScript Version: 2.2` to the standard header at the top of the file.